stream/stream_file.c

```diff
@@ -272,10 +272,14 @@ static int open_f(stream_t *stream, const struct stream_open_args *args)
     if (strncmp(url, "fd://", 5) == 0 || is_fdclose) {
         char *begin = strstr(stream->url, "://") + 3, *end = NULL;
         p->fd = strtol(begin, &end, 0);
-        if (!end || end == begin || end[0]) {
```

Comment:
In the event where `end[0] && strncomp(end, "/p/", 3) != 0` you need to return an error, otherwise you allow `fd://0foobar` and the like.

Reply:
```c
        if (!end || end == begin || (end[0] && strncmp(end, "/", 1) != 0)) {
            MP_ERR(stream, "Invalid FD: %s\n", stream->url);
            return STREAM_ERROR;
        }
        if (strncmp(end, "/p/", 3) == 0) {
            begin = strstr(end, "/p/") + 3, end = NULL;
            filename = stream->path = begin;
        }
```

Is this ok? Close the file descriptor field with a slash to ensure its integrity, if the field after it is not supported, just ignore it.